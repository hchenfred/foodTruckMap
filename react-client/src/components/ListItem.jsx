import React, { Component } from 'react';

class ListItem extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    e.preventDefault();
    this.props.truck.marker.setAnimation(google.maps.Animation.BOUNCE);
    setTimeout(() => {
      this.props.truck.marker.setAnimation(null);
     }, 1000);
  }

  render() {
    const {applicant, address, dayshours, fooditems} = this.props.truck;
    return (
      <div onClick={this.handleClick} className="card">
        <h5 className="card-title">
          {applicant}
        </h5>  
        <p><i className="fa fa-calendar" aria-hidden="true"></i> {dayshours}</p>
        <p><i className="fa fa-cutlery" aria-hidden="true"></i> {fooditems}</p>
        <p><i className="fa fa-map-marker" aria-hidden="true"></i> {address}</p>
      </div>
    );
  }
};

export default ListItem;