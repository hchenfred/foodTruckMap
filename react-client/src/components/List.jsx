import React from 'react';
import ListItem from './ListItem.jsx';

const List = (props) => (
  <section className="list-group list">
      {props.searchedResults.map((truck, index) => {
       return  <ListItem truck={truck} key={truck.applicant + truck.cnn}/>
      }
    )}
  </section>
);

export default List;