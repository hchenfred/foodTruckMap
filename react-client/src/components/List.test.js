import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import List from './List';
import ListItem from './ListItem';

test('List component should render as expected', () => {
  const searchedResults = [{
    applicant: 'Munch A Bunch',
    address: '100 Brannan St',
    dayshours: 'Mo-Fr:10AM-11AM',
    fooditems: 'good food!',
    cnn: '123456'
  }];
  const component = shallow(<List searchedResults={searchedResults}/>);
  const tree = toJson(component);
  expect(component.find(ListItem).length).toBe(1);
  expect(tree).toMatchSnapshot();
});

test('List component should render as expected', () => {
  const component = shallow(<List searchedResults={[]}/>);
  const tree = toJson(component);
  expect(component.find(ListItem).length).toBe(0);
  expect(tree).toMatchSnapshot();
});