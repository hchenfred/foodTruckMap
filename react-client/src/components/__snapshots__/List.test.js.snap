// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render as expected 1`] = `
<section
  className="list-group list"
>
  <ListItem
    truck={
      Object {
        "address": "100 Brannan St",
        "applicant": "Munch A Bunch",
        "cnn": "123456",
        "dayshours": "Mo-Fr:10AM-11AM",
        "fooditems": "good food!",
      }
    }
  />
</section>
`;
